torch/csrc/api/src/nn/modules/linear.cpp

```diff
@@ -13,8 +13,8 @@ Linear::Linear(size_t features_in, size_t features_out)
 
 void Linear::reset() {
   weight_ =
-      register_parameter("weight", at::CPU(at::kFloat).empty({out_, in_}));
-  bias_ = register_parameter("bias", at::CPU(at::kFloat).empty(out_));
+      register_parameter("weight", at::empty({out_, in_}));
```

Comment:
These changes needed to the C++ API are a good sentinel for how much we will be breaking people, and what core functions we should maintain backward compatibility for.